### DISCUSSION TITLE
Please use a gs:// URL fro your Firebase Storage bucket. - Android (Emulator & Device)

### DISCUSSION THREAD

**Discussion by Canser** [invertase/react-native-firebase#4899] [Q&A] [answered]
Hi everybody.

When I upload a file using iOS the storage works fine.
But when I try to upload something using an Android device; throws this message: (both using an emulator or a physical device)

> Please use a gs:// URL fro your Firebase Storage bucket.

The basic functionality example has taken from the official documentation page.

```
     const reference = storage().ref(image.filename);
     await reference.putFile(image.path);
     return await reference.getDownloadURL();
```

Version: 

```
"@react-native-firebase/app": "^10.3.0",
"@react-native-firebase/storage": "^10.3.1",
```

Am I missing something here?

<img width="382" alt="Screen Shot 2021-02-11 at 12 49 54" src="https://user-images.githubusercontent.com/3609898/107621720-c061d380-6c67-11eb-8000-d0602a5d59c9.png">

**Comment by mikehardy** [comment]
Well, I realize versions move fast, but reporting things on old versions is usually a losing proposition - I'd make sure, before reporting anything to any project, that you can reproduce it on current stable versions, which is 10.7.0 now here.

More to the point though, you can always look at live usage examples by checking the e2e suite (which runs in android devices every time we make a change that triggers continuous integration) 

This before block in particular runs successfully before every test in the enclosing describe block 
https://github.com/invertase/react-native-firebase/blob/3e980d4bb5fbaeefc045f2427a9e0cc0b153af6e/packages/storage/e2e/StorageReference.e2e.js#L88-L97

**Reply by Canser** [reply]
@mikehardy Hi thank you for your answer, I did update the packages to the latest version (10.7.0) but still, the result was the same.

Tried like below:
 ```
    const reference = firebase.storage().ref(image.filename);
    reference.writeToFile(image.path);
```

No chance...

**Reply by mikehardy** [reply]
:thinking: the E2E tests work, how do they work?

**Reply by Canser** [reply]
I really do not know, seeking an answer :D

**Reply by mikehardy** [reply]
The answer is found by following this: https://github.com/invertase/react-native-firebase/blob/master/tests/README.md and personally investigating. To be clear: we do not reproduce the problem, rather we reproduce successful execution, so the burden of investigation is on you

**Comment by vognam** [comment, marked as answer]
I had this same issue on android but not on ios. Turns out if the parameter inside `storage().ref()` is undefined it throws this error. Check that `image.filename` is not undefined.

**Reply by vivek-r-k** [reply]
Yes!! In my case too `image.filename` is undefined
`const choosePhotoFromLibrary = () => {
      ImagePicker.openPicker({
        cropping: true,
      }).then((image) => {
        console.log(image.filename);
        const imageUri = Platform.OS === 'ios' ? image.sourceURL : image.path;
        setImage(imageUri);
      });
    };`

what should i do?

**Comment by Canser** [comment]
@vognam is right. I was sending the filename as undefined. Thank you.
